backend/services/messaging/message_service.py

```diff
@@ -375,53 +376,63 @@ def send_message_after_chat(chat_from: int, chat: str, project_id: int):
-                message.from_user_id = chat_from
-                message.to_user_id = user.id
-                message.subject = f"You were mentioned in {link} chat"
-                message.message = chat
-                messages.append(dict(message=message, user=user))
+                    message = Message()
+                    message.message_type = MessageType.MENTION_NOTIFICATION.value
+                    message.project_id = project_id
+                    message.from_user_id = chat_from
+                    message.to_user_id = user.id
+                    message.subject = f"You were mentioned in {link} chat"
+                    message.message = chat
+                    messages.append(dict(message=message, user=user))
 
-            MessageService._push_messages(messages)
+                MessageService._push_messages(messages)
 
             query = """ select user_id from project_favorites where project_id = :project_id"""
-            result = db.engine.execute(text(query), project_id=project_id)
-            favorited_users = [r[0] for r in result]
+            favorited_users_results = db.engine.execute(
+                text(query), project_id=project_id
+            )
+            favorited_users = [r[0] for r in favorited_users_results]
+
+            # Notify all contributors except the user that created the comment.
+            contributed_users_results = (
+                TaskHistory.query.with_entities(TaskHistory.user_id.distinct())
+                .filter(TaskHistory.project_id == project_id)
+                .filter(TaskHistory.user_id != chat_from)
+                .filter(TaskHistory.action == TaskAction.STATE_CHANGE.name)
+                .all()
+            )
+            contributed_users = [r[0] for r in contributed_users_results]
+
+            users_to_notify = list(set(contributed_users + favorited_users))
 
-            if len(favorited_users) != 0:
+            if len(users_to_notify) != 0:
                 project_link = MessageService.get_project_link(
                     project_id, include_chat_section=True
                 )
-                # project_title = ProjectService.get_project_title(project_id)
                 messages = []
-                for user_id in favorited_users:
-
+                for user_id in users_to_notify:
                     try:
                         user = UserService.get_user_dto_by_id(user_id)
                     except NotFound:
+                        print("User not found")
                         continue  # If we can't find the user, keep going no need to fail
-
                     message = Message()
                     message.message_type = MessageType.PROJECT_CHAT_NOTIFICATION.value
                     message.project_id = project_id
+                    message.from_user_id = chat_from
                     message.to_user_id = user.id
                     message.subject = f"{chat_from} left a comment in {project_link}"
```

Comment:
We need to query the username here to avoid adding a user id on the notification subject:
![image](https://user-images.githubusercontent.com/666291/103536387-3016be80-4e71-11eb-8e9f-f59e01dca111.png)

Reply:
Updated - added user link to comment & chat notifications